pkg/network/http/gotls/lookup/internal/generate_luts.go

```diff
@@ -0,0 +1,320 @@
+// +build ignore
+
+package main
+
+import (
+	"context"
+	"debug/elf"
+	_ "embed"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-delve/delve/pkg/goversion"
+
+	"github.com/DataDog/datadog-agent/pkg/network/go/bininspect"
+	"github.com/DataDog/datadog-agent/pkg/network/go/lutgen"
+)
+
+var (
+	outFlag               = flag.String("out", "", "output Go source file path")
+	minGoVersionFlag      = flag.String("min-go", "", "min Go version")
+	testProgramFlag       = flag.String("test-program", "", "path to test program to compile")
+	maxQuickGoVersionFlag = flag.String("max-quick-go", "", "max Go version to skip inspecting revision-level point releases for")
+	archFlag              = flag.String("arch", "", "list of Go architectures")
+	packageFlag           = flag.String("package", "", "package to use when generating source")
+	sharedBuildDirFlag    = flag.String("shared-build-dir", "", "shared directory to cache Go versions")
+)
+
+// This program is intended to be called from go generate.
+// It generates the following lookup tables:
```

Comment:
So, just so I understand. In this case we're assuming that all these "properties" are immutable given a certain Go version/arch. Are there scenario in which, for example, a standard library function could be inlined or something like that, and these values could potentially could potentially drift from the pre-computed ones?

Reply:
Yes, we are assuming that these properties are constant given a Go version/arch (no matter the code or compiler flags). There might be some edge cases with compiler options where this is not the case, but it seems to work in most cases. I'll go ahead and add a comment for this.

Also, where inlining is concerned, the 3 TLS functions that are being traced aren't being inlined, at least by Go 1.17. This is because Read and Write both use `defer` in their body, which prevents a function from getting inlined. Additionally, for Close, Go uses its "cost"/"budget" mechanism to rule out inlining the function, giving the debug diagnostic `cannot inline (*Conn).Close: function too complex: cost 318 exceeds budget 80` ([more information on the inlining mechanism here](https://medium.com/a-journey-with-go/go-inlining-strategy-limitation-6b6d7fc3b1be)).